File: 0448.Find All Numbers Disappeared in an Array/bitmap.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        int n = nums.size();
        
        vector<bool> bitmap(n + 1, 0);
        for(int i: nums){
            bitmap[i] = true;
        }
        
        vector<int> ret;
        for(int i = 1; i < n + 1; ++i){
            if(!bitmap[i])
                ret.push_back(i);
        }
        return ret;
    }
};

File: 0448.Find All Numbers Disappeared in an Array/bitmap.scala
object Solution {
    import scala.collection.mutable.ArrayBuffer;
    def findDisappearedNumbers(nums: Array[Int]): List[Int] = {
        var len = nums.length;
        var bitmap = Array.fill[Boolean](len + 1)(false);
        bitmap(0) = true;
        nums.foreach(i => bitmap(i) = true);
        
        var ret = ArrayBuffer[Int]();
        var index = 0;
        bitmap.map(i => {if(!i) ret.addOne(index); index += 1;});
        ret.toList;
    }
}